src/topology/algebra/ordered.lean

```diff
@@ -180,7 +181,24 @@ instance {α : Type*} [topological_space α] [partial_order α] [orderable_topol
   conv in (_ ∨ _) { rw or.comm }; refl⟩
 
 section partial_order
-variables [topological_space α] [partial_order α] [t : orderable_topology α]
+variable [partial_order α]
+
+/--
+(Order) topology on a partial order `α` generated by the subbase of open intervals
+`(a, ∞) = { x ∣ a < x }, (-∞ , b) = {x ∣ x < b}` for all `a, b` in `α`.
+-/
+def partial_order.topology : topological_space α :=
+generate_from {s : set α | ∃ (a : α), s = {b : α | a < b} ∨ s = {b : α | b < a}}
```

Comment:
I think this is already in mathlib under the name orderable topology.

Reply:
Sorry, this is slightly different.

Reply:
Sorry, this is slightly different.

Reply:
Clearly this should be better documented. What is already in mathlib is the predicate class saying that a given topology is generated by intervals. This allows to use lemmas from this file even if the topology was defined differently. For instance the topology on real number is defined differently, but an instance of `orderable_topology` is then proved. What Kevin is adding here is a construction that tautologically satisfies this predicate. All this should be explained in the header.